src/ctap2/mod.rs

```diff
@@ -809,5 +809,440 @@ pub(crate) fn set_or_change_pin_helper(
     };
     // the callback is expecting `Result<(), AuthenticatorError>`, but `ChangeExistingPin`
     // and `SetNewPin` return the default `ClientPinResponse` on success. Just discard it.
-    callback.call(res.map(|_| ()));
+    callback.call(res.map(|_| T::from(())));
+}
+
+pub(crate) fn bio_enrollment(
+    dev: &mut Device,
+    command: BioEnrollmentCmd,
+    status: Sender<crate::StatusUpdate>,
+    callback: StateCallback<crate::Result<crate::ManageResult>>,
+    alive: &dyn Fn() -> bool,
+) -> bool {
+    let authinfo = match dev.get_authenticator_info() {
+        Some(i) => i.clone(),
+        None => {
+            callback.call(Err(HIDError::DeviceNotInitialized.into()));
+            return false;
+        }
+    };
+
+    if authinfo.options.bio_enroll.is_none()
+        && authinfo.options.user_verification_mgmt_preview.is_none()
+    {
+        callback.call(Err(AuthenticatorError::HIDError(
+            HIDError::UnsupportedCommand,
+        )));
+        return false;
+    }
+
+    let use_legacy_preview = authinfo.options.bio_enroll.is_none();
+
+    // We are not allowed to request the BE-permission using UV, so we have to skip UV
+    let mut skip_uv = authinfo.options.uv_bio_enroll != Some(true);
+    let timeout = 30 * 1000;
```

Comment:
Do we need to set a timeout? The spec says its optional. I'm a little worried that 30s might not be long enough for all users.

Reply:
Yes, I made it optional!